wireapp/wire-ios-mono
# fix: remove tmp files on terminate, backup restoration and app switch WPB-1151

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-1151" title="WPB-1151" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-1151</a>  Unencrypted Data Persists in Supposedly Temporary Directory
  </td></table>
  <br />
 








----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Temporary files were not deleted and remained in the file system in some scenarios:
- force quit
- backup restoration
- switching app

### Solutions

Added code for files removal in case of:
- app termination
- backup restoration
- app resigning being active

### Testing

#### Test Coverage (Optional)


#### How to Test

1. With usage of simulator find a directory where app's filesystem is stored on the drive.

For force quit/switching app
2. Open app and cause files to be downloaded to tmp (send or open a file attachment).
3. List files in the predetermined directory. It should contain the attachments
4. Force quit the app or switch the app
5. List files in the predetermined directory. It should be empty.

For backup restoration
2. Log in into the app and import backup
3. After success, list files in the predetermined directory. It should be empty.


----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
